crates/typst-cli/src/compile.rs

```diff
@@ -335,6 +354,55 @@ impl ExportCache {
     }
 }
 
+/// Writes a Makefile rule describing the relationship between `output` and
+/// `dependencies` to `makefile_deps_path`.
+fn create_makefile_deps(
+    makefile_deps_path: &Path,
+    output: &Path,
+    dependencies: impl Iterator<Item = PathBuf>,
+) -> io::Result<()> {
+    // Based on `munge` in libcpp/mkdeps.cc from the GCC source code. This isn't
+    // perfect as some special characters can't be escaped.
+    fn munge(p: &Path) -> String {
+        let s = p.to_string_lossy();
+        let mut res = String::with_capacity(s.len());
+        let mut slashes = 0;
+        for c in s.chars() {
+            match c {
+                '\\' => slashes += 1,
+                '$' => {
+                    res.push('$');
+                    slashes = 0;
+                }
+                ' ' | '\t' => {
+                    // `munge`'s source contains a comment here that says: "A
+                    // space or tab preceded by 2N+1 backslashes represents N
+                    // backslashes followed by space..."
+                    res.push_str(&"\\".repeat(slashes + 1));
+                    slashes = 0;
+                }
+                '#' => {
+                    res.push('\\');
+                    slashes = 0;
+                }
+                _ => slashes = 0,
+            };
+            res.push(c);
+        }
+        res
+    }
+
+    let mut file = File::create(makefile_deps_path)?;
```

Comment:
Yeah different tools appear to do this in slightly different ways. GCC in particular has a pile of different flags to control where the output goes, how paths and escaping are handled, etc.

The only use case I can think of where someone might want to output it to stdout and redirect from there is if they're appending it to an existing file, but in that case I think it would be nicer to just output it to its own file and `include` it from the other Makefile.